### ISSUE TITLE
Exception thrown in MembershipOracle.TryToSuspectOrKill when debug level set to Debug

### ISSUE THREAD

**Issue by benjaminpetit** [dotnet/orleans#4507]
The following line is incorrect:

``` csharp
if (logger.IsEnabled(LogLevel.Debug)) logger.Debug("-TryToSuspectOrKill {0}: The current status of {0} in the table is {1}, its entry is {2}", entry.SiloAddress.ToLongString(), entry.Status, entry.ToFullString());
````

The parameter `format` does not follow the behavior of `String.Format`, so it will throw an exception "System.FormatException : Index (zero based) must be greater than or equal to zero and less than the size of the argument list".

Format should respect this format: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-2.1&tabs=aspnetcore2x#log-message-template

We need to do a pass in Orleans codebase to see if we did the same mistake in other places.

**Comment by andscho-msft** [comment]
Did this get past tests because the tests run with a higher minimum LogLevel? Perhaps we should use Trace for LogLevel in tests.

**Comment by benjaminpetit** [comment]
Yes indeed. I am trying to run them with loglovel set to trace to see if it has a big impact or not in the test time. Some of our test relies on timing, so I wouldn't be surprised if we have more test failure using this log level

**Comment by sergeybykov** [comment]
Resolved via #4508.
